src/PrestaShopBundle/Entity/Repository/AttributeGroupRepository.php

```diff
@@ -37,39 +40,42 @@
 class AttributeGroupRepository extends \Doctrine\ORM\EntityRepository
 {
     /**
-     * @param bool $withAttributes
-     * @param array $attributeIds
+     * Asserts that attribute groups exists in all the provided shops.
+     * If at least one of them is missing in any shop, it throws exception.
      *
-     * @return array
+     * @param AttributeGroupId[] $attributeGroupIds
+     * @param ShopId[] $shopIds
+     *
+     * @throws ShopAssociationNotFound
      */
-    public function listOrderedAttributeGroups(bool $withAttributes, array $attributeIds = []): array
+    public function assertExistsInEveryShop(array $attributeGroupIds, array $shopIds): void
```

Comment:
Maybe this should be moved in the other repository, especially since you use `getArrayResult` so if it's not relying on the Entity it might as well be in the ObjectModel/Connection repository instead?